dev-util/git-delta/git-delta-0.17.0.ebuild

```diff
@@ -0,0 +1,246 @@
+	memchr@2.7.1
+	miniz_oxide@0.7.1
+	mio@0.8.11
+	nix@0.27.1
+	ntapi@0.4.1
+	nu-ansi-term@0.49.0
+	num-traits@0.2.17
+	once_cell@1.19.0
+	onig@6.4.0
+	onig_sys@69.8.1
+	option-ext@0.2.0
+	palette@0.7.3
+	palette_derive@0.7.3
+	path_abs@0.5.1
+	pathdiff@0.2.1
+	percent-encoding@2.3.1
+	phf@0.11.2
+	phf_generator@0.11.2
+	phf_macros@0.11.2
+	phf_shared@0.11.2
+	pkg-config@0.3.28
+	plist@1.6.0
+	powerfmt@0.2.0
+	proc-macro2@1.0.75
+	quick-xml@0.31.0
+	quote@1.0.35
+	rand@0.8.5
+	rand_core@0.6.4
+	redox_syscall@0.4.1
+	redox_users@0.4.4
+	regex-automata@0.4.3
+	regex-syntax@0.7.5
+	regex-syntax@0.8.2
+	regex@1.10.2
+	rgb@0.8.37
+	rustix@0.38.28
+	ryu@1.0.16
+	safemem@0.3.3
+	same-file@1.0.6
+	semver@1.0.21
+	serde@1.0.194
+	serde_derive@1.0.194
+	serde_json@1.0.111
+	serde_yaml@0.9.30
+	shell-words@1.1.0
+	siphasher@0.3.11
+	smol_str@0.1.24
+	std_prelude@0.2.12
+	strsim@0.10.0
+	syn@2.0.48
+	syntect@5.1.0
+	sysinfo@0.29.11
+	termcolor@1.4.0
+	terminal-colorsaurus@0.3.1
+	terminal-trx@0.2.0
+	terminal_size@0.3.0
+	thiserror-impl@1.0.56
+	thiserror@1.0.56
+	time-core@0.1.2
+	time-macros@0.2.16
+	time@0.3.31
+	tinyvec@1.6.0
+	tinyvec_macros@0.1.1
+	unicode-bidi@0.3.14
+	unicode-ident@1.0.12
+	unicode-normalization@0.1.22
+	unicode-segmentation@1.10.1
+	unicode-width@0.1.11
+	unsafe-libyaml@0.2.10
+	url@2.5.0
+	utf8parse@0.2.1
+	vcpkg@0.2.15
+	walkdir@2.4.0
+	wasi@0.11.0+wasi-snapshot-preview1
+	wasm-bindgen-backend@0.2.89
+	wasm-bindgen-macro-support@0.2.89
+	wasm-bindgen-macro@0.2.89
+	wasm-bindgen-shared@0.2.89
+	wasm-bindgen@0.2.89
+	wild@2.2.0
+	winapi-i686-pc-windows-gnu@0.4.0
+	winapi-util@0.1.6
+	winapi-x86_64-pc-windows-gnu@0.4.0
+	winapi@0.3.9
+	windows-core@0.52.0
+	windows-sys@0.45.0
+	windows-sys@0.48.0
+	windows-sys@0.52.0
+	windows-targets@0.42.2
+	windows-targets@0.48.5
+	windows-targets@0.52.0
+	windows_aarch64_gnullvm@0.42.2
+	windows_aarch64_gnullvm@0.48.5
+	windows_aarch64_gnullvm@0.52.0
+	windows_aarch64_msvc@0.42.2
+	windows_aarch64_msvc@0.48.5
+	windows_aarch64_msvc@0.52.0
+	windows_i686_gnu@0.42.2
+	windows_i686_gnu@0.48.5
+	windows_i686_gnu@0.52.0
+	windows_i686_msvc@0.42.2
+	windows_i686_msvc@0.48.5
+	windows_i686_msvc@0.52.0
+	windows_x86_64_gnu@0.42.2
+	windows_x86_64_gnu@0.48.5
+	windows_x86_64_gnu@0.52.0
+	windows_x86_64_gnullvm@0.42.2
+	windows_x86_64_gnullvm@0.48.5
+	windows_x86_64_gnullvm@0.52.0
+	windows_x86_64_msvc@0.42.2
+	windows_x86_64_msvc@0.48.5
+	windows_x86_64_msvc@0.52.0
+	xdg@2.5.2
+	yaml-rust@0.4.5
+"
+
+inherit cargo shell-completion
+
+DESCRIPTION="A syntax-highlighting pager for git"
+HOMEPAGE="https://github.com/dandavison/delta"
+SRC_URI="
+	https://github.com/dandavison/delta/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz
+	${CARGO_CRATE_URIS}
+"
+S="${WORKDIR}/${P/git-/}"
+
+LICENSE="MIT"
+# Dependent crate licenses
+LICENSE+=" Apache-2.0 BSD LGPL-3+ MIT MPL-2.0 Unicode-DFS-2016"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~riscv"
+
+BDEPEND="virtual/pkgconfig"
+DEPEND="
+	dev-libs/libgit2:0/1.7
```

Comment:
I didn't bump it yet precisely because of this. It didn't feel like an improvement to force a downgrade of libgit2.

Reply:
I could remove the subslot and the export line but then it would use the bundled libgit2 which is version 1.7.2 as well

Reply:
https://github.com/rust-lang/git2-rs/pull/1032

Of course it is not yet released... so no one releasing new versions of their own projects can update libgit2 either.

Reply:
I get that.

What I was trying to do is to use the system library as recommended in the ebuild guide for rust. However doing this requires both the `export LIBGIT2_NO_VENDOR=1` and `libgit2-1.7.x` to be present so I pinned the subslot. git-delta-0.16.5 on my system was not linked against libgit since I had some 1.8 version installed.

I was wondering if this is even worth doing or if we could just use the vendored libgit and remove the dependency in the ebuild so it could be independently upgraded if this blocks the PR. So basically keep it as in the old ebuild.